eZ/Publish/Core/Repository/Helper/DomainMapper.php

```diff
@@ -136,15 +136,17 @@ public function buildDomainFields(array $spiFields, $contentType)
 
         $fields = array();
         foreach ($spiFields as $spiField) {
-            $fields[] = new Field(
-                array(
-                    'id' => $spiField->id,
-                    'value' => $this->fieldTypeRegistry->getFieldType($spiField->type)
-                        ->fromPersistenceValue($spiField->value),
-                    'languageCode' => $spiField->languageCode,
-                    'fieldDefIdentifier' => $fieldIdentifierMap[$spiField->fieldDefinitionId],
-                )
-            );
+            if (array_key_exists($spiField->fieldDefinitionId, $fieldIdentifierMap)) {
```

Comment:
after trying to make patch for this; we won't be able to set $field->id and $field->languageCode, so I think I'm favor patch as is, but with `isset` as proposed above.